samples: openthread: FLASH/RAM optimization

lmaciejonczyk opened the pull request:
Add configuration overlay with minimal configuration for Thread CLI singleprotocol and multiprotocol sample.

pszkotak commented on samples/openthread/cli/harness/overlay-cert.conf:
```diff
@@ -30,7 +30,6 @@ CONFIG_BT_LL_SOFTDEVICE_DEFAULT=y
 
 # Enable and configure Bluetooth LE
 CONFIG_BT=y
-CONFIG_BT_SMP=y
```
Is the change in the certification overlay intentional?

lmaciejonczyk replied:
Yes, the certification tests do not create the bluetooth connection, only advertise. This option enables support for bluetooth secure connection.

ru-fu reviewed (changes requested):
I added two commit with some small fixes. Please check and squash if okay!

We could move the commit that adds information to ug_app_dev.rst into a separate PR if you want. But would be nice to have. :)

lmaciejonczyk commented:
> I added two commit with some small fixes. Please check and squash if okay!
> 
> We could move the commit that adds information to ug_app_dev.rst into a separate PR if you want. But would be nice to have. :)

It looks good to me. I've added missing parts. Review last commit please.

ru-fu reviewed (commented):
Another small commit. :) 
Also, two comments I made yesterday but forgot to submit ... 🙄

lmaciejonczyk commented on doc/nrf/ug_app_dev.rst:
```diff
@@ -115,3 +117,46 @@ Complete the following actions to optimize the Bluetooth part of your applicatio
+* Reduce the stack sizes of the Thread internal threads where possible.
+
+  Use the :ref:`zephyr:thread_analyzer` to analyze the stack usage.
+  You should enable the following configuration options to output the current use:
```
It is general, and already described here: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/zephyr/guides/debugging/thread-analyzer.html#configuration

ru-fu replied:
And we have a link, so no need to duplicate. I'll remove. :)

lmaciejonczyk commented on doc/nrf/ug_app_dev.rst:
```diff
@@ -115,3 +117,46 @@ Complete the following actions to optimize the Bluetooth part of your applicatio
+  The following configuration options affect the stack sizes of the Thread threads:
+
+  * :option:`CONFIG_OPENTHREAD_THREAD_STACK_SIZE`
+  * :option:`CONFIG_NET_CONNECTION_MANAGER_STACK_SIZE`
```
It makes sense to me.

ru-fu reviewed (approved):
Looks good!

lmaciejonczyk commented:
@carlescufi PR is green, merge it please.

lmaciejonczyk commented:
@rlubos Can you help to merge it please?
